recognition/advanced_tree_search.py

```diff
@@ -397,14 +411,14 @@ def create_config(
             ]
 
         post_config.flf_lattice_tool.global_cache.read_only = True
-        post_config.flf_lattice_tool.global_cache.file = lm_gc.out_global_cache
+        post_config.flf_lattice_tool.global_cache.file = gc
 
-        arpa_lms = AdvancedTreeSearchLmImageAndGlobalCacheJob.find_arpa_lms(
+        arpa_lms = lm.find_arpa_lms(
```

Comment:
We already have called lm.find_arpa_lms above. Maybe move the above call out of the if condition and then reuse the `arpa_lms` from above here. This would also avoid mismatching items in the list as I outlined above.